src/WebJobs.Script.Extensibility/ScriptBindingContext.cs

```diff
@@ -107,12 +114,32 @@ public ScriptBindingContext(JObject bindingMetadata)
         public TValue GetMetadataValue<TValue>(string name, TValue defaultValue = default(TValue))
         {
             JToken value = null;
-            if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+            try
             {
-                return value.Value<TValue>();
+                if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+                {
+                    return value.Value<TValue>();
+                }
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException($"Invalid value specified for binding property '{name}' of type {PrettyTypeName(typeof(TValue))}.", e);
             }
 
             return defaultValue;
         }
+
+        private static string PrettyTypeName(Type t)
```

Comment:
I agree this is nicer to read, but still debating whether it is the right thing to do :)

Regardless a couple of suggestions here:

1. Using string interpolation will make this look a bit cleaner
2. Use `FullName` instead of `Name` so we get namespace information